src/provider/model/FinEnvironment.ts

```diff
@@ -33,6 +33,8 @@ export class FinEnvironment extends AsyncInit implements Environment {
      */
     public readonly windowClosed: Signal1<Identity> = new Signal1();
 
+    private readonly windowCreationTimes: {[id: string]: number} = {};
```

Comment:
"windows know their own creation time" - like, from the OpenFin API?

It's done this way so it's the actual creation time, rather than (creation + however long it takes for `registerWindow` to get called), and unless there is an OpenFin API, it's only FinEnvironment that knows that, as it's the only place we can sensibly listen for the `window-created` event

I was very tempted, to get rid of wrapApplication, and have windowCreated dispatch an `AppWindow`, as I /think/ this would remove the need for the map, but thought that might be controversial 😛

Reply:
✅